tools/c7n_mailer/pyproject.toml

```diff
@@ -32,15 +32,23 @@ redis = "^3.4.1"
 jsonpointer = "^2.0"
 jsonpatch = "^1.25"
 types-six = "^1.16.10"
-google-cloud-secret-manager = "^2.8.0"
 importlib-metadata = ">=4.11.1, <5.0"
 
+# Optional packages for additional provider support
+c7n-gcp = { version = "*", optional = true }
+google-cloud-secret-manager = { version = "^2.8.0", optional = true }
+c7n-azure = { version = "*", optional = true }
```

Comment:
i _think_ sendgrid can also be optional?

Reply:
Looks like it was initially that way, but #5434 made it work with AWS too.

There may be other deps we can extract over time though :+1: .